src/components/Feed.js

```diff
@@ -1,9 +1,234 @@
 
-const Feed = () => (
-    <div className="container feed">
-    </div>
-);
+
+async function getListOfPosts(AuthStr) {
+    return axios.get("http://" + API_DOMAIN + "/api/posts/", { headers: { Authorization: AuthStr }, params: { feed: true,}
+    })
+        .then(result => result.data);
+}
+
+async function getListOfComments(AuthStr, post_id) {
+    return axios.get("http://" + API_DOMAIN + "/api/comments/", { headers: { Authorization: AuthStr }, params: { post_id: post_id,}
+    })
+        .then(result => result.data);
+}
+
+async function getUser(AuthStr, user_id) {
+    const user_str = ''.concat(user_id);
+    return axios.get("http://" + API_DOMAIN + "/api/users/" + user_str + "/",
+        { headers: { Authorization: AuthStr }, params: { feed: true,}
+    })
+        .then(result => result.data);
+}
+
+function postComment(AuthStr, credentials) {
+    return axios.post("http://" + API_DOMAIN + "/api/comments/", credentials, {
+        headers: {Authorization: AuthStr}
+    }).catch((error) => {
+        console.log(error.response);
+    });
+}
+
+function postLike(AuthStr, post_id, is_like) {
+    const post_str = ''.concat(post_id);
+    if (is_like) {
+        return axios.post("http://" + API_DOMAIN + "/api/posts/" + post_str + "/like/", {}, {
+            headers: {Authorization: AuthStr}
+        }).catch((error) => {
+            console.log(error.response);
+        });
+    }
+    else {
+        return axios.post("http://" + API_DOMAIN + "/api/posts/" + post_str + "/dislike/", {}, {
+            headers: {Authorization: AuthStr}
+        }).catch((error) => {
+            console.log(error.response);
+        });
+    }
+}
+
+function deleteLike(AuthStr, post_id, is_like) {
+    const post_str = ''.concat(post_id);
+    if (is_like) {
+        return axios.delete("http://" + API_DOMAIN + "/api/posts/" + post_str + "/like/",  {
+            headers: {Authorization: AuthStr}
+        }).catch((error) => {
+            console.log(error.response);
+        });
+    }
+    else {
+        return axios.delete("http://" + API_DOMAIN + "/api/posts/" + post_str + "/dislike/", {
+            headers: {Authorization: AuthStr}
+        }).catch((error) => {
+            console.log(error.response);
+        });
+    }
+}
+
+async function getLikesCount(AuthStr, post_id) {
+    const post_str = ''.concat(post_id);
+    return axios.get("http://" + API_DOMAIN + "/api/posts/" + post_str + "/likes/count/", { headers: { Authorization: AuthStr }
+    })
+        .then(result => result.data);
+}
+
+function Feed(props) {
+    const { token, setToken } = useToken();
+    const [ postList, setPostList ] = useState([]);
+    const [ loading, setLoading ] = useState(true);
+    const AuthStr = 'Token '.concat(token);
+
+    useEffect(() => {
+        const handleSome = async () => {
+            setLoading(true);
+            const posts = await getListOfPosts(AuthStr);
+
+            let ind = 0;
+            for (let item of posts) {
```

Comment:
What is `item`? I can't get it from the first look. Readable vars names will save a lot of time maintaining the application